configHelper/configHelper/config_helper.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+import boto3
+import logging
+import json
+import yaml
+
+logging.basicConfig(format='%(asctime)s [%(levelname)s] %(message)s', level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def merge_dicts(dict1, dict2):
+    """Recursively merges two dictionaries."""
+    for key, value in dict2.items():
+        if key in dict1 and isinstance(dict1[key], dict) and isinstance(value, dict):
+            merge_dicts(dict1[key], value)
+        else:
+            dict1[key] = value
+
+def get_secret_as_dict(secret_arn, aws_region):
+    client = boto3.client('secretsmanager', region_name=aws_region)
+    response = client.get_secret_value(SecretId=secret_arn)
+
+    if 'SecretString' in response:
+        secret_string = response['SecretString']
+    else:
+        raise RuntimeError(f'SecretString not found in response when retrieving secret value: {secret_arn}')
+
+    try:
+        secret_dict = json.loads(secret_string)
+    except json.JSONDecodeError as json_error:
+        raise RuntimeError(f"Failed to parse secret string as JSON: {json_error}")
+    return secret_dict
+
+
+def modify_auth_config(yaml_dict, auth_secret_arn, aws_region, security_realm_id):
+    auth_secret_dict = get_secret_as_dict(auth_secret_arn, aws_region)
+    yaml_auth_dict = yaml_dict.get("jenkins", {}).get("securityRealm", {}).get(security_realm_id, {})
+    if not yaml_auth_dict:
+        raise RuntimeError(f'Unable to find jenkins.securityRealm.{security_realm_id} path in initial Jenkins config yaml')
```

Comment:
Sometimes if security is disabled, I believe securityRealm can be absent (not sure about this but I remember deleting everything security to disable security in my dev set up), in that case do we want to add the securityRealm instead of throwing an error?
WDYT?

Reply:
Currently we do not call this python script unless an auth type is given. We definitely could alter the python script to be able to handle say a default auth type and essentially just print a log line. I'm fine with either approach, do you have any preference? 
```
      // Make any changes to initial jenkins.yaml
      InitCommand.shellCommand(loginAuthProps.authType !== 'default'
        // eslint-disable-next-line max-len
        ? `cd /configHelper && sudo pipenv run python3 config_helper.py --jenkins-config-file-path=/initial_jenkins.yaml --auth-secret-arn=${loginAuthProps.authCredsSecretsArn} --security-realm-id=${realm} --aws-region=${stackRegion} > configHelper.log 2>&1`
        : 'No changes made to initial_jenkins.yaml with respect to AuthType'),
```

Reply:
I agree that the script is only called when auth-type is specified. Was just concerned about the edge case where the initial jenkins.yaml might be missing `securityRealm` key altogether. We can get this in and monitor how it goes.

Reply:
My expectation is that the python script would throw error and result in the `initial_jenkins.yaml` not getting updated. Curious if there is any particular way you want to handle this?

Reply:
How about add the missing key `jenkins.securityRealm` if not found?

Reply:
Pushed an update so that if we can't find the `jenkins.securityRealm` or auth object after it, we will create it